plutus-tx-plugin/src/PlutusTx/Plugin.hs

```diff
@@ -126,9 +130,22 @@ parsePluginArgs args = do
             , poDumpPir = elem "dump-pir" args
             , poDumpPlc = elem "dump-plc" args
             , poOptimize = notElem "dont-optimize" args
+            , poMaxSimplifierIterations = maxIterations
             }
     -- TODO: better parsing with failures
     pure opts
+    where
+        prefix :: String
+        prefix = "max-simplifier-iterations="
+        defaultIterations :: Int
+        defaultIterations = view PIR.coMaxSimplifierIterations PIR.defaultCompilationOpts
+        maxIterations :: Int
+        maxIterations = case filter (isPrefixOf prefix) args of
```

Comment:
:see_no_evil: makes me want to write an actual option parser :D

Reply:
this one works for now.